theano/sandbox/cuda/blocksparse.py

```diff
@@ -645,19 +645,22 @@ def local_inplace_blocksparse_outer(node):
         if node.op == sparse_block_outer_ss:
             return [sparse_block_outer_ss_inplace(*node.inputs)]
 
-    # Should be run before elemwise fusion
-    @opt.register_opt()
-    @alpha_merge(SparseBlockOuterSS, alpha_in=5, nd=4)
-    def local_merge_blocksparse_alpha(node, *inputs):
-        """
-GpuElemwise{mul}(lr, SparseBlockOuterSS) -> SparseBlockOuterSS(..., alpha=lr)
-        """
-        return [sparse_block_outer_ss(*inputs)]
-
-    @opt.register_opt()
-    @output_merge(SparseBlockOuterSS, alpha_in=5, out_in=0, nd=4)
-    def local_merge_blocksparse_output(node, *inputs):
-        return [sparse_block_outer_ss(*inputs)]
+# XXX: these optimisations were badly broken and now require a working
```

Comment:
I don't recall them giving errors, but that is probably just because we where lucky and less people use blocksparse that convolution.

They most definitely where broken before though.  The answer to your second question is on the line just below.  They now require a proper beta to work and SparseBlockOuterSS doesn't have it.

This can be fixed with some effort, but it is more than I am prepared to do right now.